:memo: Add changelog for 1.2.0 (#1323)
diasurgical/devilutionX: docs/CHANGELOG.md

## DevilutionX 1.2.0 (unreleased)
### Features
- Hellfire support
- Shareware support
- Fully migrate saves between Diablo and Hellfire

> - Support file paths longer then 259 characters

Reviewer: ```suggestion
- Support file paths longer than 259 characters
```

> - Support file paths longer than 259 characters

Author: ```suggestion
- Document [ini-file options](https://github.com/diasurgical/devilutionX/wiki)
- Support file paths longer than 259 characters
```

- Allow for up to 99 save games of each type
- Much lower memory usage (now in line with original)

> - Performance improvments

Reviewer: ```suggestion
- Performance improvements
```

> - Performance improvments

Author: ```suggestion
- Performance improvements
```

> - Gold picked goes to stack even when inventroy is open

Reviewer: ```suggestion
- Gold picked goes to stack even when inventory is open
```

> - Gold picked goes to stack even when inventroy is open

Author: ```suggestion
- Gold picked goes to stack even when inventory is open
```

- Pepin automatically heals
- Reorganize ini-file
- Show other players on the map
- Widescreen menus

> - Scroll subtitles at the same speed as naration reader

Reviewer 2: ```suggestion
- Scroll subtitles at the same speed as narration reader
```

- Preserve list position when buy and selling items in stores
- Use belt via numpad

> - Close panels when pressing esc

Reviewer: ```suggestion
- Close panels when pressing <kbd>ESC</kbd>
```

- Shift-click to spend all stat points
- Shift-click to clear redied spell
- Shift-click consumables to move them between the belt and inventory
- Shift-click equipment eqipt/unequip them

> - Preserve hokeys and active spell across games

Reviewer: ```suggestion
- Preserve hotkeys and active spell across games
```

> - Set default active spell for new heros

Author: ```suggestion
- Set default active spell for new heroes
```

> - Apply sound volumn to videos

Reviewer: ```suggestion
- Apply sound volume to videos
```

- DIABDAT.MPQ can now be in upper case
- Improved gamepad support:
  - D-Pad / Left stick navigation now works in all menus and is more responsive.
  - START + SELECT now opens the main menu.
  - Right stick mouse emulation has been improved.
  - Gamepads plugged in while the game is running can be used.
  - All connected gamepads can be used (previously, only the first one).

> - Add `--ttf-dir` and `-ttf-name` to allow for specifying a differen UI font

Reviewer: ```suggestion
- Add `--ttf-dir` and `--ttf-name` to allow for specifying a different UI font
```

### Optional improvments

> - Improved transparancy

Reviewer: ```suggestion
- Improved transparency
```

- Experience bar
- Enemy health bar
- Auto equip all item types
- Auto pickup gold
- Run in town (also in multiplayer)
- Disable friendly fire
- Show monster type
- Disable walking sounds
- No quest randomization
- Adria refils mana

> - Barbarian and Bard heros in Diablo

Author: ```suggestion
- Barbarian and Bard heroes in Diablo
```

- Controller mapping
- Configurable network port
- Support for portable installs

### Platforms
- Switch Windows releases to 64bit
- Added support for PlayStation Vita
- Added support for Nintendo 3DS
- Added support for DragonFly BSD
- Added AppImage for Linux

### Bugfixes
- Incorrect graphics under cave doors in multiplayer
- Catacombs doors never having traps
- Dragging on touch devices
- Mouse not releasing when exiting the window
- Multiplayer host crashing after beating the game
- Minor color issues in some videos
- Command-line output missing on Windows
- Correct visuals and function of the joining-game-dialog
- Mouse jumping to the top of the windows when using the menu

### Original Diablo bugs
- All bugs fixed in Hellfire 1.01
- Some unique items morphing on a new game
- Monsters being unable to properly see the location of firewalls
- Fix more stability issues
- Gold stacks are correctly topped when in dungeon

> - Gold icons graphics is always correct

Reviewer 2: ```suggestion
- Gold icons graphics are always correct
```

- Arrows graphics not aligning with the move direction
- Monsters miss calculating charge directions for some angles
- Randomly being teleported from level 15 to Lazarus chamber
- Wrong death sound when monster/golem kill another monster
- Scavengers/Gravediggers taking damage when eating on higher difficulties
- Not getting XP from monsters hit by golems

> - Sneaks fleeing in the wrong direction of golem

Author: ```suggestion
- Hiddens fleeing in the wrong direction of golem
```

> - Hiddens fleeing in the wrong direction of golem

Author: ```suggestion
- Monsters with hiding ability fleeing in the wrong direction from golems
```

- Monsters going inactive when fighting golems offscreen
- Gharbad the weak not dropping his item when killed by a golem

> - Lights not some times not properly following unique monsters

Reviewer 2: ```suggestion
- Lights not following unique monsters after reentering the level
```

Author: ```suggestion
- Lights not always following unique monsters
```

- Several bugs in Lazarous quest when re-entering the teleport
- Lights added on map change not working (Arkaine's Valor etc)
- Player light radius not being following when a player is knocked back
- Player light radius not updating properly fully before changing level
- Tiles not properly being marked as out of visibility when walking
- Quest items brought from previous games breaking the game state
- Inconsistent repair prices caused by rounding errors
- Several issues causing Arkaine's Valor to enter a broken state
- Base damage not being updated when leveling up
- Flash only dealing 10% damage in some directions
- Town portals appearing on two levels when cast in a quest level
- Incorrectly seeing legit items as duplicates
- Gaining mana from reading books despite wearing an item with corruption
- Fallen fleeing in the wrong direction
- Not getting XP for the first monster placed on a level
- Loadscreen colors on entering the Poison Water quest

> - Being able to waist gold on red scrolls when not having room for them

Reviewer 2: ```suggestion
- Being able to waste gold on red scrolls when not having room for them
```

- Inconsistent mouse behavior when opening/closing various panels
- Items being destroyed if held in cursor when entering dungeon
- Black Mushroom quest not resetting between games
- Lava pools sometimes having roof tiles in them
- Doors sometimes missing in Halls of The Blind

### Original Hellfire bugs
- All bugs fixed between Diablo 1.04 and 1.09
- Monster attack sound sometimes not playing
- Firewall damage calculation being too low, and different for the first player in multiplayer

> - Missing sounds when bard and barbarians talks to cows

Reviewer 2: ```suggestion
- Missing sounds when bards and barbarians talk to cows
```

- Right-hand item's decay suffix being applied to the left-hand item in multiplayer
- Town Shrine would always cast TP from the host instead of the user
- Oily Shrine and Sparkling Shrine would cast their traps as if the first player was doing it
- The farmer would always think he was talking to the local player

> - Scavenger/Gravedigger AI being stuck for around after eating

Reviewer 2: ```suggestion
- Scavenger/Gravedigger AI being stuck for a round after eating
```

- Minor typography corrections
- Sync avalible quests in multiplayer
- Correct spell level being ignored for Search and Rage

> - "0 Gold" somtimes spawning on The Cornerston

Reviewer: ```suggestion
- "0 Gold" sometimes spawning on The Cornerstone
```

## DevilutionX 1.2.0 (unreleased)
### Features
- Hellfire support
- Shareware support
- Fully migrate saves between Diablo and Hellfire
- Document [ini-file options](https://github.com/diasurgical/devilutionX/wiki)
- Support file paths longer than 259 characters
- Allow for up to 99 save games of each type
- Much lower memory usage (now in line with the original)
- Performance improvements

> - Gold picked goes to stack even when inventory is open

Reviewer 3: Actually didn't know we implemented this 😆

Author: Well technically it's brought over from Hellfire :)

- Pepin automatically heals
- Reorganize ini-file
- Show other players on the map
- Widescreen menus
- Scroll subtitles at the same speed as narration reader
- Preserve list position when buy and selling items in stores
- Use belt via the numpad
- Close panels when pressing <kbd>ESC</kbd>
- Shift-click to spend all stat points
- Shift-click to clear readied spell
- Shift-click consumables to move them between the belt and inventory
- Shift-click equipment equip/unequip them
- Preserve hotkeys and active spell across games
- Set default active spell for new heroes
- Apply sound volume to videos
- DIABDAT.MPQ may now be upper case
- Improved gamepad support:
  - D-Pad / Left stick navigation now works in all menus and is more responsive.
  - START + SELECT now opens the main menu.
  - Right stick mouse emulation has been improved.
  - Gamepads plugged in while the game is running can be used.
  - All connected gamepads can be used (previously, only the first one).
- Add `--ttf-dir` and `--ttf-name` to allow for specifying a different UI font

### Optional improvements
- Improved transparency
- Experience bar

> - Enemy health bar

Reviewer 3: Suggest we refer to "monsters" in a consistent manner
```suggestion
- Monster health bar
```

Reviewer 4: What about multiplayer pvp and party members?

> - Auto equip all item types

Reviewer 3: Would suggest adding "on pickup" to avoid potential confusion
```suggestion
- Auto equip all item types on pickup
```

Author: But it also happens when buying them :D

Reviewer: ```suggestion
- Auto equip all item types on pickup or purchase
```

Reviewer 3: > But it also happens when buying them :D

🤣 I implemented it and forgot about it.

> - Auto equip all item types on pickup or purchase

Author: ```suggestion
- Auto equip all item types on pickup or purchase
- Auto-equip sounds
```

- Auto pickup gold
- Run in town (also in multiplayer)
- Disable friendly fire
- Show monster type
- Disable walking sounds
- No quest randomization
- Adria refills mana
- Barbarian and Bard heroes in Diablo
- Controller mapping
- Configurable network port
- Support for portable installs

> 

Reviewer 3: + Auto-equip sounds

Author: Done

### Platforms
- Switch Windows releases to 64bit
- Added support for PlayStation Vita
- Added support for Nintendo 3DS
- Added support for DragonFly BSD
- Added AppImage for Linux

### Bugfixes
- Incorrect graphics under cave doors in multiplayer
- Catacombs doors never having traps
- Dragging on touch devices
- Mouse not releasing when exiting the window
- Multiplayer host crashing after beating the game
- Minor color issues in some videos
- Command-line output missing on Windows
- Correct visuals and function of the joining-game-dialog
- Mouse jumping to the top of the windows when using the menu

### Original Diablo bugs
- All bugs fixed in Hellfire 1.01
- Some unique items morphing on a new game
- Monsters being unable to properly see the location of firewalls
- Fix more stability issues

> - Gold stacks are correctly topped when in dungeon

Reviewer 3: What does this mean?

Author: The stack is toped to 5000/10000 before a new one is started, instead of 4813 when the next pickup is 188 etc.

Let me know if you can come up with a better way to word it :)

Author: ```suggestion
- Gold piles will be filled to their max, before starting a new one
```

- Gold icons graphics are always correct
- Arrows graphics not aligning with the move direction
- Monsters miss calculating charge directions for some angles
- Randomly being teleported from level 15 to Lazarus chamber

> - Wrong death sound when monster/golem kill another monster

Reviewer 3: ```suggestion
- Wrong death sound when monster/golem kills another monster
```

- Scavengers/Gravediggers taking damage when eating on higher difficulties
- Not getting XP from monsters hit by golems
- Monsters with hiding ability fleeing in the wrong direction from golems
- Monsters going inactive when fighting golems offscreen

> - Gharbad the weak not dropping his item when killed by a golem

Reviewer 3: ```suggestion
- Gharbad the Weak not dropping his item when killed by a golem
```

- Lights not always following unique monsters

> - Several bugs in Lazarous quest when re-entering the teleport

Reviewer 3: ```suggestion
- Several bugs in Lazarus quest when re-entering the teleport
```

- Lights added on map change not working (Arkaine's Valor etc)

> - Player light radius not being following when a player is knocked back

Reviewer 3: ```suggestion
- Player light radius not readjusting when a player is knocked back
```
(original seems a bit weird to me)

Author: I think "being" was a leftover from another edit :)

> - Player light radius not updating properly fully before changing level

Reviewer 3: ```suggestion
- Player light radius not updating properly before changing level
```

## DevilutionX 1.2.0 (unreleased)
### Features
- Hellfire support
- Shareware support
- Fully migrate saves between Diablo and Hellfire

> - Document [ini-file options](https://github.com/diasurgical/devilutionX/wiki)

Reviewer 5: ```suggestion
- Document [ini-file options](https://github.com/diasurgical/devilutionX/wiki/DevilutionX-diablo.ini-configuration-guide)
```

- Support file paths longer than 259 characters
- Allow for up to 99 save games of each type
- Much lower memory usage (now in line with the original)
- Performance improvements
- Gold picked goes to stack even when inventory is open
- Pepin automatically heals
- Reorganize ini-file
- Show other players on the map
- Widescreen menus
- Scroll subtitles at the same speed as narration reader

> - Preserve list position when buy and selling items in stores

Reviewer 6: ```suggestion
- Preserve list position when buying and selling items in stores
```

- Shift-click to clear readied spell
- Shift-click consumables to move them between the belt and inventory
- Shift-click equipment equip/unequip them
- Preserve hotkeys and active spell across games
- Set default active spell for new heroes
- Apply sound volume to videos
- DIABDAT.MPQ may now be upper case
- Improved gamepad support:
  - D-Pad / Left stick navigation now works in all menus and is more responsive.
  - START + SELECT now opens the main menu.
  - Right stick mouse emulation has been improved.
  - Gamepads plugged in while the game is running can be used.
  - All connected gamepads can be used (previously, only the first one).
- Add `--ttf-dir` and `--ttf-name` to allow for specifying a different UI font

### Optional improvements
- Improved transparency
- Experience bar
- Monster health bar
- Auto equip all item types on pickup or purchase
- Auto-equip sounds
- Auto pickup gold
- Run in town (also in multiplayer)
- Disable friendly fire
- Show monster type
- Disable walking sounds
- No quest randomization
- Adria refills mana
- Barbarian and Bard heroes in Diablo
- Controller mapping
- Configurable network port
- Support for portable installs

### Platforms

> - Switch Windows releases to 64bit

Reviewer 6: ```suggestion
- Switch Windows releases to 64-bit
```

Reviewer 5: Are we switching the release to 64-bit or adding 64 in addition to 32?

Author: I think we might be switching. A slight bit undecided, but I don't see many 32bit machines running xp+ at this point.

- Added support for PlayStation Vita
- Added support for Nintendo 3DS
- Added support for DragonFly BSD
- Added AppImage for Linux

### Bugfixes
- Incorrect graphics under cave doors in multiplayer
- Catacombs doors never having traps
- Dragging on touch devices
- Mouse not releasing when exiting the window
- Multiplayer host crashing after beating the game
- Minor color issues in some videos
- Command-line output missing on Windows
- Correct visuals and function of the joining-game-dialog
- Mouse jumping to the top of the windows when using the menu

### Original Diablo bugs
- All bugs fixed in Hellfire 1.01
- Some unique items morphing on a new game
- Monsters being unable to properly see the location of firewalls
- Fix more stability issues
- Gold piles will be filled to their max, before starting a new one
- Gold icons graphics are always correct
- Arrows graphics not aligning with the move direction
- Monsters miss calculating charge directions for some angles

> - Randomly being teleported from level 15 to Lazarus chamber

Reviewer 6: ```suggestion
- Randomly being teleported from level 15 to Lazarus' chamber
```

- Wrong death sound when monster/golem kills another monster
- Scavengers/Gravediggers taking damage when eating on higher difficulties
- Not getting XP from monsters hit by golems
- Monsters with hiding ability fleeing in the wrong direction from golems
- Monsters going inactive when fighting golems offscreen
- Gharbad the Weak not dropping his item when killed by a golem
- Lights not always following unique monsters
- Several bugs in Lazarus quest when re-entering the teleport

> - Lights added on map change not working (Arkaine's Valor etc)

Reviewer 6: ```suggestion
- Lights added on map change not working (Arkaine's Valor, etc)
```

Author: ```suggestion
- Lights added on map update not working (Arkaine's Valor, etc.)
```

- Player light radius not readjusting when a player is knocked back
- Player light radius not updating properly before changing level
- Tiles not properly being marked as out of visibility when walking
- Quest items brought from previous games breaking the game state
- Inconsistent repair prices caused by rounding errors
- Several issues causing Arkaine's Valor to enter a broken state
- Base damage not being updated when leveling up
- Flash only dealing 10% damage in some directions
- Town portals appearing on two levels when cast in a quest level
- Incorrectly seeing legit items as duplicates
- Gaining mana from reading books despite wearing an item with corruption
- Fallen fleeing in the wrong direction
- Not getting XP for the first monster placed on a level
- Loadscreen colors on entering the Poison Water quest
- Being able to waste gold on red scrolls when not having room for them
- Inconsistent mouse behavior when opening/closing various panels
- Items being destroyed if held in cursor when entering dungeon
- Black Mushroom quest not resetting between games
- Lava pools sometimes having roof tiles in them

> - Doors sometimes missing in Halls of The Blind

Author: ```suggestion
- Doors sometimes missing in Halls of The Blind
- Broken outline on some inventory items
```
